src/fs_fd.ts

```diff
@@ -292,6 +327,11 @@ export class OpenDirectory extends Fd {
       0,
     ).ret;
   }
+
+  path_unlink_file(path: string): number {
+    delete this.dir.contents[path];
+    return wasi.ERRNO_SUCCESS;
```

Comment:
Could you add a FIXME to check if the path actually exists and to return an error otherwise. Also is there no test in wasmtime's test suite for that?